src/Native/Unix/System.Native/pal_time.c

```diff
@@ -167,22 +167,22 @@ int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
     }
 
     uint64_t resolution = SystemNative_GetTimestampResolution();
-    uint64_t timestamp = SystemNative_GetTimestamp();
+    double timestamp = SystemNative_GetTimestamp();
 
-    uint64_t currentTime = timestamp * SecondsToNanoSeconds / resolution;
+    double currentTime = timestamp * SecondsToNanoSeconds / resolution;
 
     uint64_t lastRecordedCurrentTime = previousCpuInfo->lastRecordedCurrentTime;
     uint64_t lastRecordedKernelTime = previousCpuInfo->lastRecordedKernelTime;
     uint64_t lastRecordedUserTime = previousCpuInfo->lastRecordedUserTime;
 
-    uint64_t cpuTotalTime = 0;
+    double cpuTotalTime = 0;
     if (currentTime > lastRecordedCurrentTime)
     {
         // cpuTotalTime is based on clock time. Since multiple threads can run in parallel,
         // we need to scale cpuTotalTime cover the same amount of total CPU time.
         // rusage time is already scaled across multiple processors.
         cpuTotalTime = (currentTime - lastRecordedCurrentTime);
-        cpuTotalTime *= (uint64_t)numProcessors;
+        cpuTotalTime *= (double)numProcessors;
```

Comment:
isn't there a chance of overflow here as well ?

Reply:
For the first iteration the lastRecordedtime is zero

Reply:
The original case can be solved because there is a division after the multiplication, so you may get back within range even if the multiplication overflows.

You are out of luck here, however, since nothing can scale the result back in range if the multiplication causes it to overflow.